src/hooks/use-click-outside/__tests__/index.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { useRef } from 'react'
+import { useClickOutside } from '..'
+import { fireEvent, render } from '@testing-library/react'
+
+const TestComponent = ({ onClickOutside, isUsingParent = true }) => {
+  const ref = useRef<HTMLDivElement>(null)
+  useClickOutside(ref, onClickOutside)
+
+  if (isUsingParent) {
+    return (
+      <div>
+        <div ref={ref}>Inside Element</div>
+      </div>
+    )
+  } else {
+    return <div ref={ref}>Inside Element</div>
+  }
+}
+
+describe('useClickOutside', () => {
+  const onClickOutside = jest.fn()
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const renderElement = () =>
+    render(
+      <>
+        <TestComponent onClickOutside={onClickOutside} />
+        <div>Outside Element</div>
+      </>,
+    )
+
+  describe('has parent ', () => {
+    it('should call onClickOutside when clicking outside the element', () => {
+      const { getByText } = renderElement()
+
+      fireEvent.mouseDown(getByText('Outside Element'))
```

Comment:
**question:** Currently there is no element with this text in it, so I'm not sure how this test is passing 🤔 If it is passing, that would suggest we're getting a false positive and we need to adjust the test so that this scenario correctly fails (otherwise our test isn't designed to "fail well" and its value is undermined)

Reply:
that text actually under `renderElement` fn, yeah I think it could cause false positive, I will update the test to be more clear 👍